### PR TITLE
allow running terraform-resources based integrations per account

### PR DESCRIPTION
**Pull request by BumbleFeng**
Signed-off-by: Feng Huang <fehuang@redhat.com>

### CODE REVIEW

**Comment by tparikh** [comment]
Nice work @BumbleFeng. @maorfr @jmelis PTAL

File: reconcile/terraform_resources.py [hunk 1]
```diff
@@ -238,9 +238,13 @@ def init_working_dirs(accounts, thread_pool_size,
     return ts, working_dirs
 
 
-def setup(print_only, thread_pool_size, internal, use_jump_host):
+def setup(print_only, thread_pool_size, internal, use_jump_host,
+          account_name):
     gqlapi = gql.get_api()
     accounts = queries.get_aws_accounts()
+    if account_name:
+        accounts = [n for n in accounts
```
Comment (maorfr) [inline reconcile/terraform_resources.py:246]: should we check that `accounts` is not empty?

**Review by maorfr — APPROVED** [review]
LGTM

very nice! is this going to be instead of sharding or in addition?
i can see how this change would make it simpler for our idea of qontract-operator (instead of counting accounts and using that for shards, we just run the integration per account). is this is the intention?

File: reconcile/terraform_resources.py (hunk 1 above)
Comment (BumbleFeng) [reply reconcile/terraform_resources.py:246]: If it runs `--account-name` with nothing, get `Error: --account-name option requires an argument`.
If it runs with `--account-name ""`, it runs for all accounts.

Comment (BumbleFeng) [reply reconcile/terraform_resources.py:246]: If no account matches, it only runs the oc part.

Comment (maorfr) [reply reconcile/terraform_resources.py:246]: Can we avoid it by checking?

Comment (BumbleFeng) [reply reconcile/terraform_resources.py:246]: Sure, raise an error for account not found

**Pull request merged**